src/Servers/Kestrel/Transport.NamedPipes/src/Internal/NamedPipeConnectionListener.cs

```diff
@@ -0,0 +1,181 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.IO.Pipelines;
+using System.IO.Pipes;
+using System.Net;
+using System.Threading.Channels;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.Extensions.Logging;
+using NamedPipeOptions = System.IO.Pipes.PipeOptions;
+using PipeOptions = System.IO.Pipelines.PipeOptions;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.NamedPipes.Internal;
+
+internal sealed class NamedPipeConnectionListener : IConnectionListener
+{
+    private readonly ILogger _log;
+    private readonly NamedPipeEndPoint _endpoint;
+    private readonly NamedPipeTransportOptions _options;
+    private readonly CancellationTokenSource _listeningTokenSource = new CancellationTokenSource();
+    private readonly CancellationToken _listeningToken;
+    private readonly Channel<ConnectionContext> _acceptedQueue;
+    private readonly MemoryPool<byte> _memoryPool;
+    private readonly PipeOptions _inputOptions;
+    private readonly PipeOptions _outputOptions;
+    private readonly Mutex _mutex;
```

Comment:
It seems this is never used

Reply:
Ah nvm, it is used for cross-process protection, I see it from the connection factory